src/checkstyle/checks/SpacingCheck.hx

```diff
@@ -41,8 +48,19 @@ class SpacingCheck extends Check {
 					if (post) dist = e.pos.max - e2.pos.max;
 					else dist = e2.pos.min - e.pos.min;
 					if (dist > unopSize(uo)) logPos('Space around ${unopString(uo)}', e.pos, Reflect.field(SeverityLevel, severity));
-				case EIf(econd, eif, eelse) if (spaceIfCondition):
-					if (econd.pos.min - e.pos.min < "if (".length) logPos('No space between if and condition', e.pos, Reflect.field(SeverityLevel, severity));
+				case EIf(econd, _, _) if (spaceIfCondition):
+					checkSpaceBetweenExpressions('if', e, econd);
+				case EFor(it, _) if (spaceForLoop):
+					checkSpaceBetweenExpressions('for', e, it);
+				case EWhile(econd, _, true) if (spaceWhileLoop):
```

Comment:
Only handling regular `while` loops here for now, `do...while` is a bit trickier.